src/Bicep.Cli.IntegrationTests/BuildParamsCommandTests.cs

```diff
@@ -110,8 +110,40 @@ public async Task Build_Params_With_CLI_Input_And_Using_Declaration_Bicep_File_R
             result.Should().Be(1);
             output.Should().BeEmpty();
             error.Should().Contain($"Bicep file {otherBicepPath} provided with --bicep-file option doesn't match the Bicep file {bicepPath} referenced by the using declaration in the parameters file");
+            // error
         }
 
+
+        [TestMethod]
+        public async Task Build_Params_Bicep_File_Reference_Mismatch_And_Other_Diagnostics_ShouldFail_WithAllExpectedErrorMessages()
+        {
+            var bicepparamsPath = FileHelper.SaveResultFile(TestContext, "input.bicepparam", 
+            @"
+            using './main.bicep'
+
+            param foo = 'bar'
+            ");
+            var bicepPath = FileHelper.SaveResultFile(TestContext, "main.bicep", "param foo object", Path.GetDirectoryName(bicepparamsPath));
+
+            var otherBicepPath = FileHelper.SaveResultFile(TestContext, "otherMain.bicep", "", Path.GetDirectoryName(bicepparamsPath));
+
+            var settings = new InvocationSettings(new(TestContext, ParamsFilesEnabled: true), BicepTestConstants.ClientFactory, BicepTestConstants.TemplateSpecRepositoryFactory);
+            var outputFilePath = FileHelper.GetResultFilePath(TestContext, "output.json");
+
+            File.Exists(outputFilePath).Should().BeFalse();
+            var(output, error, result) = await Bicep(settings, "build-params", bicepparamsPath,"--bicep-file", otherBicepPath, "--outfile-params", outputFilePath);
+
+            var diagnostics = await GetAllParamDiagnostics(bicepparamsPath, BicepTestConstants.ClientFactory, BicepTestConstants.TemplateSpecRepositoryFactory);
+
+
+            result.Should().Be(1);
+            output.Should().BeEmpty();
+            error.Should().Contain($"Bicep file {otherBicepPath} provided with --bicep-file option doesn't match the Bicep file {bicepPath} referenced by the using declaration in the parameters file");
+            error.Should().ContainAll(diagnostics);
+        }
```

Comment:
Worth adding an assertion to verify we didn't actually write to the `outputFilePath`?